### PR TITLE
Feat/Gigamon Application Insights

### PR DESCRIPTION
**Pull request by nr-jbiggley**
# Summary

A concise description of the changes being introduced. Please review the pre-merge checklist section to validate this pull request is ready for review and merge. **If it is not ready, please mark the pull request as a draft.**

## Pre merge checklist

- [Y] Did you check you NRQL syntax? - Does it work?
- [Y]Did you include an InstallPlan and Documentation reference?
- [Y]Are all documentation links publically accessible?
- [Y]Did you check your descriptive content for [voice and tone](https://docs.newrelic.com/docs/style-guide/writing-strategies/voice-strategies-docs-sound-new-relic/)? 
- [Y]Did you check your descriptive content for spelling and grammar errors?

### Dashboards 

- [Y]Does the PR contain a screenshot for each of your dashboards?
- [Y]Do your screenshots show data?
- [Y]Has the [sanitization script](https://github.com/newrelic/newrelic-quickstarts/blob/main/CONTRIBUTING.md#dashboards) been run on each dashboard?

### Alerts

- [N/A] Did you check that your alerts actually work?

### CODE REVIEW

**Comment by nr-jbiggley** [comment]
@jcountsNR After lots of confusion, I _finally_ got a PR with just the Gigamon Application Insights changes. Note that I will be updating the link to the documentation when it is published by Gigamon.

**Comment by aswanson-nr** [comment]
Looks like we are waiting on a documentation link. I'll wait for @jcountsNR approval before merging

**Comment by brammerl** [comment]
Hello, We are no longer accepting install plans as a part of the quickstart schema. We are instead moving to use data sources. You can either refer to an [already existing data source](https://github.com/newrelic/newrelic-quickstarts/tree/main/data-sources) in this repository or[ make your own](https://github.com/newrelic/newrelic-quickstarts/blob/main/CONTRIBUTING.md#data-sources-1) community data source. 

If this applies to your PR, please make the proper update to your quickstart's config to reference only data sources.

File: data-sources/gigamon-appinsights/config.yml
Comment (aswanson-nr) [inline data-sources/gigamon-appinsights/config.yml:26]: We can remove a few of these since we are going to put them on the quickstart instead!
```suggestion
  - security
  - packet capture
  - network
```

File: data-sources/gigamon-appinsights/config.yml
Comment (aswanson-nr) [inline data-sources/gigamon-appinsights/config.yml:28]: This can also be removed!
```suggestion
```

File: quickstarts/gigamon-appinsights/config.yml
Comment (aswanson-nr) [inline quickstarts/gigamon-appinsights/config.yml:28]: Then we can add those keywords over here
```suggestion
  - intelligence
  - metadata
  - siem
  - newrelic partner
  - NR1_addData
  - NR1_sys
```

File: data-sources/gigamon-appinsights/config.yml
Comment (aswanson-nr) [inline data-sources/gigamon-appinsights/config.yml:15]: Since the quickstart is actually going to be displayed instead of this data source, let's update it to do the same thing as the install plan (we're swapping out install plans for data sources very soon)
```suggestion
install:
  primary:
    link:
      url: https://www.gigamon.com/products/optimize-traffic/application-intelligence.html
```

File: quickstarts/gigamon-appinsights/config.yml
Comment (aswanson-nr) [inline quickstarts/gigamon-appinsights/config.yml:32]: The system isn't a huge fan of `.html` if you remove that, it should be fine with this link.
```suggestion
    url: www.gigamon.com/products/optimize-traffic/application-intelligence/application-metadata
```

**Review by aswanson-nr — CHANGES_REQUESTED** [review]
Hey @nr-jbiggley, This quickstart looks great! I just have a few comments around the data source and quickstart config. We can actually just display the quickstart instead of the data source and change the data source to be an underlying installation detail. The data source will eventually fully replace the install plan. 
I added some comments on what to change to allow for this, please let me know if you run into any issues!

**Comment by nr-jbiggley** [comment]
> Hey @nr-jbiggley, This quickstart looks great! I just have a few comments around the data source and quickstart config. We can actually just display the quickstart instead of the data source and change the data source to be an underlying installation detail. The data source will eventually fully replace the install plan. I added some comments on what to change to allow for this, please let me know if you run into any issues!

Awesome feedback @aswanson-nr -- I would have been foolish to not accept them all :D

File: dashboards/gigamon-appinsights/gigamon-appinsights.json
Comment (aswanson-nr) [inline dashboards/gigamon-appinsights/gigamon-appinsights.json:1980]: The validation does not allow for blank widget fields, could we remove this since it doesn't have anything?
```suggestion
```

File: quickstarts/gigamon-appinsights/config.yml
Comment (aswanson-nr) [inline quickstarts/gigamon-appinsights/config.yml:35]: I realized it's looking for a full URL here, so adding the `https://` should make it happy
```suggestion
    url: https://www.gigamon.com/products/optimize-traffic/application-intelligence/application-metadata
```

**Review by aswanson-nr — CHANGES_REQUESTED** [review]
Hey @nr-jbiggley, sorry to request more changes but we had a few CI failures for this PR

**Comment by nr-jbiggley** [comment]
> Hey @nr-jbiggley, sorry to request more changes but we had a few CI failures for this PR

All good! Committed both recommendations.

**Comment by nr-jbiggley** [comment]
@jcountsNR @aswanson-nr I've updated the PR with the correct URL from Gigamon. This is ready to be merged and deployed.

**Comment by nr-jbiggley** [comment]
Pinging @jcountsNR and @aswanson-nr for an ETA on approval and merge.

File: quickstarts/gigamon-appinsights/config.yml
Comment (brammerl) [inline quickstarts/gigamon-appinsights/config.yml:43]: You don't need to include this field if there are no alert policies for your quickstart

File: dashboards/gigamon-appinsights/gigamon-appinsights.json
Comment (brammerl) [inline dashboards/gigamon-appinsights/gigamon-appinsights.json:1975]: Seems like this is causing the dashboard workflows to fail. Can you remove the trailing comma?

Comment (nr-jbiggley) [reply dashboards/gigamon-appinsights/gigamon-appinsights.json:1975]: Fixed

**Review by nr-jbiggley — COMMENTED** [review]
@aswanson-nr Was there anything left outstanding on this PR?

**Comment by brammerl** [comment]
Hi! Seems like the dashboard json is still running into issues formatting wise. Anyway you could run it through a linter/remake it?

**Comment by seemantk** [comment]
@nr-jbiggley hi there, hello.  The file `dashboards/gigamon-appinsights/gigamon-appinsights.json` is returning as invalid json in the automated tests for this PR.  Can you please give that a once over and lint/validate it locally?

File: dashboards/gigamon-appinsights/gigamon-appinsights.json
Comment (josephgregoryii) [inline dashboards/gigamon-appinsights/gigamon-appinsights.json:1968]: Looks like we're missing a comma here!
```suggestion
            ],
```

File: data-sources/gigamon-appinsights/config.yml
Comment (aswanson-nr) [inline data-sources/gigamon-appinsights/config.yml:4]: Looks like this description is a bit long, we need to cut it down to under 240 characters.

**Review by aswanson-nr — COMMENTED** [review]
Getting this json to be valid

File: data-sources/gigamon-appinsights/config.yml
Comment (nr-jbiggley) [reply data-sources/gigamon-appinsights/config.yml]: Cut down the description

**Comment by nr-jbiggley** [comment]
> Old PRs will be closed after 30 days of inactivity. This PR has been quiet for 14 days and is being marked as stale. Reply here to keep this PR open.

Made the changes as requested @aswanson-nr -- apologies on the delay in responding!

**Comment by aswanson-nr** [comment]
Looks like we just need @brammerl to sign the CLA and it should be ready to merge!

**Comment by brammerl** [comment]
Sorry! just signed and going to merge now.

**Review by brammerl — APPROVED** [review]

**Review by aswanson-nr — APPROVED** [review]

**Pull request merged**
